packages/tpetra/core/src/Tpetra_replaceDiagonalCrsMatrix_def.hpp

```diff
@@ -90,6 +90,14 @@ replaceDiagonalCrsMatrix (CrsMatrix<SC, LO, GO, NT>& matrix,
         "Row map of matrix and map of input vector do not match.");
   }
 
+  // FENCE REVIEW - NOW PASSING
+  //   Testing: This code is exercised by unit tests.
+  //   GTX960:  Passed with CUDA_LAUNCH_BLOCKING=0 and fence removed.
+  //   White:   Passed with CUDA_LAUNCH_BLOCKING=0 and fence removed.
+  //   Plan:    Probably keep fence or refactor.
+  //   Notes:   colMap.getLocalElement should fail for UVM use of glMap_.
+  //            Need to verify if that path is tested.
+
```

Comment:
This fence is necessary for UVM. Views used in the row map and colmap can use UVM and they are accessed in the following routine. So, we need to make sure that the values are available for touching in host.

Reply:
Kept and added comments #7546